### PR TITLE
Chapter 14 exercises complete, all tests pass

### PR DESCRIPTION
**Pull request by thruddle**

### CODE REVIEW

File: drjt/My-List-Util/lib/My/List/Util.pm [hunk 1]
```diff
@@ -0,0 +1,131 @@
+package My::List::Util;
+
+use 5.010;
+use strict;
+use warnings;
+
+use List::Util qw( sum shuffle );
```
Comment (utopia-2) [inline drjt/My-List-Util/lib/My/List/Util.pm:7]: It's meant to be `My::List::Util`, not `My::Proxy::Onto::The::Real::List::Util`.

File: drjt/Animal/lib/LivingCreature.pm [hunk 2]
```diff
@@ -41,13 +41,15 @@ if you don't export anything, such as for a purely object-oriented module.
 =cut
 
 sub speak {
-  my $class = shift;
+  my ( $out, $class ) = ( undef, shift );
```
Comment (SimonGross) [inline drjt/Animal/lib/LivingCreature.pm:44]: This reads a bit strangely, as you're defining a new variable on the same line as retrieving a parameter. I'd do these on separate rows.

File: drjt/My-List-Util/lib/My/List/Util.pm (hunk 1 above)
Comment (thruddle) [inline drjt/My-List-Util/lib/My/List/Util.pm:7]: The book does suggest looking at `List::Util` for an implementation. But yeah, originally I was planning on just using `reduce` for the sum, but figured that these functions were better suited for the job.

File: drjt/My-List-Util/t/shuffle.t [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+my @out = My::List::Util::shuffle( @in );
+# and you shake it all about
+
+is( scalar @in, scalar @out, 'input and output are same length' );
+
+for my $i ( @in ) {
+  ok( $i ~~ @out, "$i is included in \@out" );
+}
+
+isnt( "@in", "@out", 'test input and output are different' ); # intermittent fail - has a 1/3628800 chance of failing
```
Comment (SimonGross) [inline drjt/My-List-Util/t/shuffle.t:23]: I'm not enthused by a deliberately coded chance of an intermittent failure, even if it is rare enough that'd you'd likely see intermittent failures for a whole host of other reasons before ever seeing this.

**Comment by SimonGross** [comment]
The `My-List-Util-0.01` directory seems to make up half of the new files you added. Is there any way to have Git avoid these?

File: drjt/My-List-Util/t/shuffle.t (hunk 3 above)
Comment (thruddle) [inline drjt/My-List-Util/t/shuffle.t:23]: Yeah, but aside from leaving it untested, there doesn't seem to be a way around it. We can minimize the fail rate by increasing list size, or doing multiple comparisons in an or, but it doesn't really seem worth it when the rate is already less than 0.0001%

**Pull request merged**